### ISSUE TITLE
Cannot copy a picture from an album to a subalbum

### ISSUE THREAD

**Issue by RodeBaert** [LycheeOrg/Lychee-front#161]
I try to copy pictures from an album to a subalbum but I'm not able to.
When I click the move button the only albums I see are the "mainalbums" and not the subalbums"

### Output of the diagnostics (Settings => Diagnostics)

    Diagnostics
    -----------
    Warning: Dropbox import not working. dropbox_key is empty.
    Warning: You may experience problems when uploading a large amount of photos. Take a look in the FAQ for details.

    System Information
    ------------------
    Lychee-front Version:      -
    Lychee Version (git):      No git data found. Probably installed from release or could not read .git
    DB Version:                040000
    System:                    Linux
    PHP Version:               7.3
    MySQL Version:             5.7.24
    
    Lychee total space:        287.45 MB
    Upload folder space:       0.00 B
    System total space:        1.20 TB
    System free space:         137.88 GB (11%)
    
    Imagick:                   1
    Imagick Active:            1
    Imagick Version:           1684
    GD Version:                bundled (2.1.0 compatible)

    Config Information
    ------------------
    version:                   040000
    check_for_updates:         1
    sorting_Photos_col:        takestamp
    sorting_Photos_order:      ASC
    sorting_Albums_col:        description
    sorting_Albums_order:      DESC
    imagick:                   1
    skip_duplicates:           0
    small_max_width:           0
    small_max_height:          360
    medium_max_width:          1920
    medium_max_height:         1080
    lang:                      en
    layout:                    1
    image_overlay:             1
    image_overlay_type:        desc
    default_license:           none
    compression_quality:       90
    full_photo:                1
    delete_imported:           1
    Mod_Frame:                 0
    Mod_Frame_refresh:         30000
    landing_page_enable:       0
    landing_owner:             PhotoLychee Team
    landing_title:             PhotoLychee Team
    landing_subtitle:          just a testwebsite
    landing_facebook:          
    landing_flickr:            
    landing_twitter:           
    landing_instagram:         
    landing_youtube:           
    landing_background:        dist/cat.jpg
    thumb_2x:                  1
    small_2x:                  0
    medium_2x:                 0
    site_title:                Lychee v4
    site_copyright_enable:     1
    site_copyright_begin:      2019
    site_copyright_end:        2019
    public_search:             0
    additional_footer_text:    
    display_social_in_gallery: 0
    public_recent:             0
    recent_age:                1
    public_starred:            0
    downloadable:              0
    photos_wraparound:         1

### Browser and system

This happens with Chrome and Edge. 
My site is installed on a shared host: www.photolychee.nl

**Comment by kamil4** [comment]
It's a subtle bug that may be triggering in your case because of the peculiarities of your shared hosting environment as we've established earlier, although I'll check in a regular 32-bit environment later if I can see the same behavior there, just to be sure.

The bug is a side effect of a behavior of Lychee v4 that I don't particularly like, namely that we sometimes -- but not always -- send IDs as strings and other times as numbers. I believe we should consistently send data to the front end as strings, which would avoid any such issues, whatever their source.

I just prototyped such a fix at your site; the changes are to `app/Album.php` and `app/Photo.php`. It seems to fix the particular problem you reported, but can you play around to see if it breaks anything else in the process?

**Comment by RodeBaert** [comment]
Hi Kamil,

Thank you very much for looking into this.
I tested a few foto's and did not notice any strange behavoviour or other side effects.
I copied the 2 files to another installation and also there it is working great.

I will test it further and will report in case of errors.

**Comment by kamil4** [comment]
I tested in a 32-bit environment and I could not reproduce it, so it must be again a peculiarity of the MySQL installation at your shared hosting. Still, as I had said, we were being unnecessarily sloppy on the server side, so I'm fixing that. The PR that I submitted for merging is more comprehensive than the version I prototyped in your environment but I wouldn't worry about that if what you have works for you.
